### PR TITLE
[SMALLFIX] Fix reading an empty file for some UFSs

### PR DESCRIPTION
**Pull request by madanadit**
When reading an empty file, opening an existing file at pos = 0 would through an Exception.

### CODE REVIEW

**Comment by madanadit** [comment]
@ifcharming @calvinjia Please take a look.

File: underfs/gcs/src/main/java/alluxio/underfs/gcs/GCSInputStream.java [hunk 1]
```diff
@@ -78,7 +74,11 @@
     mKey = key;
     mClient = client;
     mPos = pos;
-    mObject = mClient.getObject(mBucketName, mKey, null, null, null, null, mPos, null);
+    if (mPos > 0) {
```
Comment (ifcharming) [inline underfs/gcs/src/main/java/alluxio/underfs/gcs/GCSInputStream.java]: can you please add a comment to explain why `mPos=0` can not be used?

File: underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AUnderFileSystem.java [hunk 2]
```diff
@@ -409,11 +409,6 @@ protected String getRootKey() {
 
   @Override
   protected InputStream openObject(String key, OpenOptions options) throws IOException {
-    try {
-      return new S3AInputStream(mBucketName, key, mClient, options.getOffset());
-    } catch (AmazonClientException e) {
-      LOG.error("Failed to open file: {}", key, e);
-      return null;
-    }
+    return new S3AInputStream(mBucketName, key, mClient, options.getOffset());
```
Comment (calvinjia) [inline underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AUnderFileSystem.java]: AmazonClientException is a runtime exception, is it ok to throw that from this method?

Comment (madanadit) [inline underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AUnderFileSystem.java]: My bad. Thanks for the catch.

**Comment by ifcharming** [comment]
Please fix the failing test...

**Pull request merged**